runtime-sdk/src/crypto/signature/sr25519.rs

```diff
@@ -55,6 +51,33 @@ impl PublicKey {
         digest.update(message);
         let transcript = context.hash256(digest);
 
+        self.verify_transcript(transcript, signature)
+    }
+
+    /// Verify a signature.
+    pub fn verify_raw(
+        &self,
+        context: &[u8],
+        message: &[u8],
+        signature: &Signature,
+    ) -> Result<(), Error> {
+        // Convert the context to a Sr25519 SigningContext.
+        let context = schnorrkel::signing_context(context);
+        let transcript = context.bytes(message);
+        self.verify_transcript(transcript, signature)
+    }
+
+    /// Verify a signature using the given transcript.
+    pub fn verify_transcript<T: SigningTranscript>(
```

Comment:
Signature verification is Base64 encoded? So... I have to base64 encode it in Solidity before passing into verification function?

Reply:
Huh? Where did you see base64 encoding?

Reply:
Oh my bad, it was from `impl From<&'static str> for PublicKey {
    fn from(s: &'static str) -> PublicKey {
        PublicKey::from_bytes(&BASE64_STANDARD.decode(s).unwrap()).unwrap()
    }
}`